server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -89,11 +121,20 @@ class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit])
         resHeader.setHeadersFrom(serverResponse)
         resHeader.handleContentLengthAndChunkedHeaders(Option(chunkedFile.length()))
         resHeader.handleCloseAndKeepAliveHeaders(req)
-
         ctx.write(resHeader)
         // HttpChunkedInput will write the end marker (LastHttpContent) for us.
         ctx.writeAndFlush(new HttpChunkedInput(chunkedFile), channelPromise).closeIfNeeded(req)
       },
+      reactiveStreamHandler = (channelPromise, publisher) => {
+        val res: DefaultStreamedHttpResponse =
+          new DefaultStreamedHttpResponse(req.protocolVersion(), HttpResponseStatus.valueOf(serverResponse.code.code), publisher)
+
+        res.setHeadersFrom(serverResponse)
+        res.handleContentLengthAndChunkedHeaders(None)
```

Comment:
will this work if the content-length is explicitly set by the user in headers?

Reply:
I'll check if there's a test for this, but my first thought is we don't want users to set content-length when streaming. It's either transfer-encoding: chunked (stream) or explicit content-length. (4.4 https://greenbytes.de/tech/webdav/rfc2616.html#rfc.section.4.4)

Reply:
well the user might know what the length of the streaming body is going to be upfront. There's even a test for it: https://github.com/softwaremill/tapir/blob/c3f45cbbd38830eb41c41cddde0ca8f283c6efb5/server/tests/src/main/scala/sttp/tapir/server/tests/ServerStreamingTests.scala#L37, though it doesn't check if the response is chunked or not. Maybe some implementations don't allow setting conent-lenght on streaming bodies 🤔 This should really be under a test flag, not an `if`  in the test

Reply:
If the specification always requires chunked response to be without the content-length header, then this `if` is IMO correct, this shouldn't depend on implementation, right?

Reply:
How we produce the data (streaming / non-streaming) doesn't have to directly translate to chunked/non-chunked encoding. We can have content with known length, and just produce it lazily